src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -5208,8 +5267,15 @@ def _make_onedeploy_request(params):
         response_body = None
         if poll_async_deployment_for_debugging:
             logger.info('Polling the status of async deployment')
+            if params.track_deployment.lower() == 'true':
+                # once deploymentstatus/latest is available, we can use it to track the deployment
+                deployment_id = _get_latest_deployment_id(params.cmd, params.resource_group_name, params.webapp_name, deployment_status_url, params.slot)
+                if deployment_id is None:
+                    raise CLIError("Unable to fetch deployment id for this deployment")
+                deploymentstatusapi_url = _build_deploymentstatus_url(params, deployment_id)
+                _track_deployment_using_deploymentstatus_api(params.cmd, deploymentstatusapi_url, params.timeout)
             response_body = _check_zip_deployment_status(params.cmd, params.resource_group_name, params.webapp_name,
```

Comment:
we should probably not poll the old endpoint again, if we're tracking using the new one. No?

Reply:
Yes, updating this